src/card.rs

```diff
@@ -246,6 +384,56 @@ impl CardConfig {
         ret
     }
 
+    /// Return the current card configuration with new board cards.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use postflop_solver::*;
+    ///
+    /// let oop_range = "66+,A8s+,A5s-A4s,AJo+,K9s+,KQo,QTs+,JTs,96s+,85s+,75s+,65s,54s";
+    /// let ip_range = "QQ-22,AQs-A2s,ATo+,K5s+,KJo+,Q8s+,J8s+,T7s+,96s+,86s+,75s+,64s+,53s+";
+    /// let ranges = [oop_range.parse().unwrap(), ip_range.parse().unwrap()];
+    ///
+    /// let card_config = CardConfig {
+    ///     range: ranges,
+    ///     flop: flop_from_str("Td9d6h").unwrap(),
+    ///     turn: card_from_str("Qc").unwrap(),
+    ///     river: NOT_DEALT,
+    /// };
+    ///
+    /// let cards = cards_from_str("Th9d3c4h").unwrap();
+    /// let card_config2 = card_config.with_cards(cards).unwrap();
+    /// assert_eq!(card_config2.range, ranges);
+    /// assert_eq!(card_config2.flop, [34, 29, 4]);
+    /// assert_eq!(card_config2.turn, 10);
+    /// assert_eq!(card_config2.river, NOT_DEALT);
+    /// ```
+    pub fn with_cards(&self, cards: Vec<Card>) -> Result<CardConfig, String> {
```

Comment:
I personally find this function a bit unintuitive. I would've expected `with_cards` to consume `self` instead of taking it as reference, and also to update `flop`/`turn`/`river` no matter the previous number of dealt cards. IIUC what's actually happening is that a copy of the config is made, keeping only `range`.

It would make more sense to me if we had `CardConfig::from_range_and_cards(ranges, cards)`, which essentially did the same thing, but required the caller to explicitly reference `old_config.cards` when constructing the new config.

Reply:
`Blah::with_blarg` is typically immutable. To modify a `Blah` I'd use a `Blah::set_blarg`.

I'll update the doc string to reflect that.

Yeah, I wasn't sure what to do if the number of cards wasn't the same. It kinda makes no sense to take a turn range and apply it to the flop, but maybe someone wants to do that? But I feel like FlopCardConfigs, TurnCardConfigs, and RiverCardConfigs should have different types (like, not actually).

If you _really_ want a flop range w/ a flop+turn you can construct it directly.

Happy to discuss further though! I'm really not sure!